src/can.c

```diff
@@ -168,10 +168,16 @@ static void thingset_can_addr_claim_rx_cb(const struct device *dev, struct can_f
             THINGSET_CAN_SOURCE_GET(frame->id), data[0], data[1], data[2], data[3], data[4],
             data[5], data[6], data[7]);
 
-    if (ts_can->node_addr == THINGSET_CAN_SOURCE_GET(frame->id)) {
+    uint8_t source_addr = THINGSET_CAN_SOURCE_GET(frame->id);
+
+    if (ts_can->node_addr == source_addr) {
         k_event_post(&ts_can->events, EVENT_ADDRESS_ALREADY_USED);
     }
 
+    if (ts_can->addr_claim_callback != NULL) {
```

Comment:
I feel like this merits an `else`; if the address is already in use, there's no point rebroadcasting it.

Reply:
This rx callback now serves two different purposes:
1. Find out whether another node already uses the desired CAN node address (only happening during claiming of its own address).
2. Forward all received address claim messages from other nodes.

For the second case, if during address discovery all already claimed nodes broadcast their node address, I don't think it makes sense to exclude the one that just happens to have the address we desire ourselves. So I think there should not be an `else`, as these two applications are independent of each other.